packages/generator/src/templateHelpers.ts

```diff
@@ -112,9 +127,7 @@ export const isObjectProperty = function(property: any): boolean {
 
 const isPrimitive = function(range: any): boolean {
   return (
-    range !== undefined &&
-    range.dataType !== undefined &&
-    range.dataType.value !== undefined
+    range != null && range.dataType != null && range.dataType.value() != null
```

Comment:
We do not have to execute the value() function to determine if it is primitive..Checking for presence if good enough...

Reply:
there are cases where value is also null. Example: offset